src/index.ts

```diff
@@ -135,16 +163,40 @@ class Compiler {
     }
   }
 
-  private createTsType(rule: JSONSchema.Schema): TsType.TsType {
+  private createTsType (rule: JSONSchema.Schema, propName?: string, isTop: boolean = false, isReference: boolean = false): TsType.TsType {
     switch (this.getRuleType(rule)) {
       case RuleType.AnonymousSchema:
       case RuleType.NamedSchema:
         return this.toTsDeclaration(rule)
       case RuleType.Enum:
-        return new TsType.Union(uniqBy(
-          rule.enum!.map(_ => this.toStringLiteral(_))
-          , _ => _.toType(this.settings))
-        )
+        // TODO:  honor the schema's "type" on the enum.  if string,
+        // skip all the zipping mess; if int, either require the tsEnumNames 
+        // or generate literals for the values
+
+        // TODO:  what to do in the case where the value is an Object?  
+        // right now we just pring [Object object] as the literal which is bad
```

Comment:
This is the only real concern I have about the code here.  I'll call attention to the example of this on the enum test case.